File: medium/cpp/2214.cpp
// https://leetcode.com/problems/minimum-health-to-beat-game/

class Solution
{
public:
    long long minimumHealth(vector<int>& damage, int armor)
    {
        auto s = std::accumulate(std::cbegin(damage), std::cend(damage), 0LL);

        auto res = s + 1;

        for (auto d : damage)
        {
            if (armor >= d)
            {
                res = std::min(res, s - d + 1);
            }
            else
            {
                res = std::min(res, s - armor + 1);
            }
        }

        return res;
    }
};

File: medium/python/2214.py
# https://leetcode.com/problems/minimum-health-to-beat-game/description/

class Solution:
    def minimumHealth(self, damage: List[int], armor: int) -> int:
        total, worst = 0, 0

        for d in damage:
            total += d
            worst = max(worst, d)

        return 1 + total - min(worst, armor)        